web-ui/src/helpers/query-parameters.js

```diff
@@ -21,8 +21,21 @@ import { useEffect } from 'react';
 /**
  * @param {(QPBoolean | QPString)[]} qps - query parameters
  */
-export const useQueryParameters = qps => {
+export const useQueryParameters = (
+  qps,
+  requirements = [],
+  processedQPs = null
+) => {
+  // This examines all the query parameters and
+  // sets the state value associated with each of them.
   useEffect(() => {
+    if (processedQPs?.current) return;
```

Comment:
If the page supplied a value for `processedQPs` AND the value it holds is `true` then we don't need to run the code that follows again.